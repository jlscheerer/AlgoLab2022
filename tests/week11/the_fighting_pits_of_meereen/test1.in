30
69 2 2
1
1
0
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
1
1
1
1
0
0
0
0
1
1
0
0
1
1
1
1
1
0
0
0
0
0
1
1
1
1
1
1
1
1
1
1
0
0
0
1
1
1
1
0
0
1
1
1
0
0
1
1
0
57 2 2
0
0
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
0
1
1
1
1
1
0
0
1
1
1
0
0
0
1
1
1
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
0
1
74 2 2
1
0
0
0
1
1
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
0
0
0
1
1
1
0
0
0
1
1
0
0
1
1
0
0
0
0
0
1
1
1
1
1
1
73 2 2
0
0
1
1
1
0
0
0
1
1
0
0
0
1
1
0
0
0
0
0
0
0
0
0
0
1
1
0
0
0
0
0
1
1
0
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
1
1
1
1
1
1
1
1
1
1
0
0
1
1
1
1
1
1
0
0
135 2 2
0
1
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
0
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
1
1
0
0
1
1
0
0
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
0
1
85 2 2
1
0
0
1
1
0
0
0
1
1
0
0
0
1
1
0
0
1
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
0
0
1
1
1
1
1
0
0
0
1
1
1
1
0
0
0
1
1
0
0
0
1
1
1
0
0
1
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
0
0
82 2 2
1
0
0
0
0
0
1
1
1
1
1
1
1
0
0
0
0
0
1
1
0
0
0
0
1
1
0
0
0
1
1
1
1
1
0
0
0
1
1
0
0
0
1
1
0
0
0
1
1
1
1
0
0
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
1
0
0
0
1
1
1
0
0
0
45 2 2
1
0
0
1
1
1
0
0
0
0
1
1
1
1
1
0
0
0
1
1
0
0
0
0
0
0
0
0
1
1
0
0
0
1
1
0
0
0
1
1
1
1
1
0
0
79 2 2
0
1
1
0
0
0
1
1
1
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
0
0
1
1
0
0
1
1
1
0
0
0
1
1
0
0
1
1
1
0
0
0
1
1
1
0
0
0
1
1
0
0
0
0
1
1
1
0
0
0
1
1
1
37 2 2
1
1
0
0
1
1
0
0
1
1
0
0
1
1
0
0
1
1
1
1
1
1
1
1
1
0
0
1
1
0
0
0
1
1
0
0
0
35 2 2
0
0
0
0
1
1
0
0
0
1
1
1
1
0
0
1
1
0
0
0
0
1
1
1
0
0
1
1
1
1
0
0
1
1
0
168 2 2
0
0
1
1
0
0
0
0
1
1
1
1
1
0
0
1
1
0
0
0
1
1
1
1
0
0
0
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
1
1
0
0
0
1
1
0
0
1
1
0
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
1
0
0
1
1
1
0
0
0
1
1
1
0
0
1
1
1
0
0
0
0
0
0
1
1
0
0
0
1
1
0
0
0
1
1
1
0
0
1
1
1
1
1
0
0
1
1
0
0
0
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
1
1
1
1
1
1
1
0
0
0
0
0
1
1
1
1
1
1
1
1
1
7 2 2
1
1
0
0
0
0
1
19 2 2
1
0
0
1
1
0
0
1
1
1
0
0
1
1
1
1
1
1
1
156 2 2
1
1
1
1
1
1
1
1
1
0
0
1
1
1
1
0
0
1
1
1
1
0
0
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
0
1
1
1
1
1
1
0
0
1
1
0
0
0
0
1
1
0
0
0
0
0
0
0
0
0
1
1
0
0
1
1
0
0
0
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
0
1
1
0
0
0
0
1
1
0
0
0
0
0
0
0
1
1
1
1
0
0
1
1
1
0
0
0
0
0
1
1
1
0
0
0
0
0
1
1
0
0
0
0
0
0
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
66 2 2
1
0
0
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
1
1
0
0
0
0
1
1
1
1
1
1
1
0
0
0
0
0
1
1
1
0
0
0
1
1
1
1
1
0
0
0
0
0
0
0
1
1
0
0
1
1
1
0
0
1
1
109 2 2
1
0
0
1
1
1
1
1
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
0
1
1
0
0
0
1
1
1
1
1
1
0
0
0
1
1
1
1
0
0
1
1
1
0
0
0
1
1
1
1
1
1
1
0
0
0
0
1
1
0
0
1
1
1
0
0
0
0
1
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
1
1
1
1
156 2 2
1
1
1
1
1
0
0
0
1
1
0
0
0
0
1
1
1
1
0
0
0
0
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
1
1
1
1
0
0
0
0
0
0
0
1
1
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
0
0
1
1
1
1
1
1
0
0
0
0
0
0
0
0
0
1
1
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
1
1
1
1
1
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
1
0
0
1
1
1
0
0
1
1
1
0
0
0
1
1
1
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
0
0
1
107 2 2
0
1
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
1
1
1
1
1
0
0
0
1
1
1
0
0
1
1
1
0
0
1
1
1
0
0
0
0
0
0
0
0
1
1
0
0
0
1
1
0
0
0
0
1
1
1
0
0
0
1
1
1
0
0
1
1
1
1
1
0
0
1
1
0
0
0
1
1
0
0
0
0
0
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
0
28 2 2
0
0
1
1
0
0
0
0
1
1
1
1
1
1
1
1
0
0
1
1
1
1
1
1
1
0
0
0
67 2 2
0
1
1
0
0
0
1
1
0
0
1
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
1
1
0
0
1
1
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
0
0
0
1
1
0
0
0
0
0
1
1
69 2 2
0
1
1
1
1
0
0
0
0
0
1
1
0
0
1
1
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
1
0
0
1
1
1
0
0
0
0
1
1
0
0
0
1
1
1
0
0
1
1
1
1
1
0
0
0
1
181 2 2
0
1
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
1
1
0
0
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
0
0
1
1
1
0
0
0
1
1
0
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
0
0
0
0
1
1
1
1
0
0
1
1
0
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
0
0
1
1
1
1
0
0
1
1
0
0
0
1
1
1
1
0
0
0
0
0
1
1
1
1
0
0
0
1
1
0
0
1
1
1
1
0
0
0
1
1
1
1
0
0
1
187 2 2
1
1
1
0
0
1
1
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
0
0
1
1
1
1
0
0
0
0
1
1
1
1
1
1
1
1
1
1
1
1
0
0
0
1
1
1
0
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
1
1
1
1
0
0
0
0
1
1
1
1
1
0
0
0
1
1
0
0
0
1
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
1
1
1
1
0
0
0
0
1
1
0
0
1
1
0
0
1
1
0
0
1
1
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
0
0
0
1
1
1
0
0
1
1
1
0
0
123 2 2
0
0
1
1
0
0
0
0
1
1
0
0
1
1
0
0
0
1
1
1
1
0
0
0
0
1
1
0
0
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
1
0
0
1
1
1
0
0
0
1
1
1
0
0
0
0
0
1
1
1
1
1
1
0
0
0
0
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
0
1
1
1
0
0
1
1
1
1
0
0
1
1
0
0
1
1
1
1
1
0
0
1
1
1
1
0
0
0
1
1
0
0
1
1
198 2 2
1
1
0
0
0
1
1
0
0
1
1
0
0
1
1
1
1
1
0
0
0
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
1
1
0
0
1
1
1
0
0
0
1
1
0
0
0
1
1
1
1
1
0
0
0
0
1
1
0
0
0
0
1
1
1
0
0
1
1
1
1
1
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
1
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
1
1
1
1
1
0
0
1
1
1
1
1
1
0
0
1
1
0
0
1
1
1
1
0
0
1
1
1
0
0
1
1
0
0
0
1
1
0
0
1
1
0
0
1
1
1
1
0
0
0
1
1
0
0
1
1
1
0
0
0
1
1
1
1
0
0
1
1
0
0
0
66 2 2
1
1
1
1
1
1
1
0
0
1
1
0
0
1
1
0
0
1
1
0
0
1
1
0
0
0
1
1
0
0
0
0
0
1
1
1
0
0
0
0
0
1
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
1
0
0
1
1
0
0
1 2 2
1
67 2 2
0
0
0
1
1
1
0
0
1
1
0
0
0
1
1
1
1
1
0
0
1
1
0
0
1
1
1
0
0
0
0
1
1
0
0
0
1
1
0
0
1
1
1
1
1
0
0
0
0
0
0
0
0
1
1
1
0
0
0
1
1
0
0
0
1
1
0
197 2 2
0
1
1
0
0
0
0
1
1
1
1
1
0
0
1
1
1
0
0
1
1
0
0
0
0
0
0
1
1
1
0
0
0
0
1
1
0
0
0
0
1
1
0
0
1
1
1
0
0
1
1
1
0
0
0
1
1
1
1
0
0
0
0
0
1
1
1
1
0
0
1
1
1
1
0
0
1
1
1
0
0
1
1
1
0
0
1
1
1
1
0
0
0
0
0
0
0
1
1
1
1
1
0
0
0
0
1
1
0
0
0
1
1
0
0
0
0
0
1
1
1
1
0
0
1
1
1
1
0
0
0
1
1
1
1
1
0
0
0
0
0
1
1
1
1
1
1
0
0
0
0
1
1
1
1
1
0
0
1
1
0
0
1
1
0
0
1
1
0
0
0
0
1
1
0
0
1
1
0
0
1
1
0
0
0
0
0
1
1
0
0
1
1
0
0
1
1
