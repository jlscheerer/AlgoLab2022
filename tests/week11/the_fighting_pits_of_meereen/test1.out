109896
93914
124889
112890
238797
145872
130877
69932
135881
62944
57947
277748
9989
29971
249766
104901
179836
254766
177839
39958
111899
112896
304728
312719
204815
339703
111901
998
109899
321704
